Modules/Microsoft365DSC/DSCResources/MSFT_EXOEmailTenantSettings/MSFT_EXOEmailTenantSettings.psm1

```diff
@@ -0,0 +1,450 @@
+        [ValidateSet('Yes')]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.Boolean]
+        $EnablePriorityAccountProtection,
+
+        [Parameter()]
+        [System.Boolean]
+        $IsValid,
+
+        [Parameter()]
+        [System.String]
+        $Identity,
+
+        [Parameter()]
+        [System.String]
+        $Name,
+
+        [Parameter()]
+        [System.String]
+        $ObjectState,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [System.String]
+        $CertificatePath,
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $CertificatePassword,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens
+    )
+    Write-Verbose -Message 'Getting EXO Email Tenant Settings'
+
+    if ($Global:CurrentModeIsExport)
+    {
+        $ConnectionMode = New-M365DSCConnection -Workload 'ExchangeOnline' `
+            -InboundParameters $PSBoundParameters `
+            -SkipModuleReload $true
+    }
+    else
+    {
+        $ConnectionMode = New-M365DSCConnection -Workload 'ExchangeOnline' `
+            -InboundParameters $PSBoundParameters
+    }
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace 'MSFT_', ''
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $nullReturn = $PSBoundParameters
+    $nullReturn.Ensure = 'Absent'
+
+    try
+    {
+        $EmailTenantSettings = Get-EmailTenantSettings -ErrorAction Stop
+
+        $result = @{
+            IsSingleInstance                           = 'Yes'
+            Identity                                   = $EmailTenantSettings.Identity
+            EnablePriorityAccountProtection            = $EmailTenantSettings.EnablePriorityAccountProtection
+            Name                                       = $EmailTenantSettings.Name
+            IsValid                                    = $EmailTenantSettings.IsValid
+            ObjectState                                = $EmailTenantSettings.ObjectState
+            Credential                                 = $Credential
+            Ensure                                     = 'Present'
```

Comment:
If there is a single instance of this on the tenant, let's remove the Ensure property. Normally resources exposing the IsSingleInstance property should not have the Ensure parameter.